apps/simple-trading-app-e2e/src/integration/market-trade.test.ts

```diff
@@ -0,0 +1,67 @@
+import { aliasQuery } from '@vegaprotocol/cypress';
+import { generateSimpleMarkets } from '../support/mocks/generate-markets';
+import { generateDealTicket } from '../support/mocks/generate-deal-ticket';
+
+describe('Market trade', () => {
+  let markets;
+  beforeEach(() => {
+    cy.mockGQL((req) => {
+      aliasQuery(req, 'SimpleMarkets', generateSimpleMarkets());
+      aliasQuery(req, 'DealTicketQuery', generateDealTicket());
+    });
+    cy.visit('/markets');
+    cy.wait('@SimpleMarkets').then((response) => {
+      if (response.response?.body?.data?.markets?.length) {
+        markets = response.response?.body?.data?.markets;
```

Comment:
Chore (non-blocking): There's a couple of things that could clean this up a bit -
1. You could destructure on line 13 and remove the need for one of the 'responses' if you'd like
2. Because you've used the elvis operator on line 14 to check existence, you don't need to use it at all on line 15

Reply:
yeah, too many operators there. Fixed